news/tests/test_forms.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+from django.db.models import Q
+from news import forms
+
+
+@pytest.mark.django_db
+class TestNewsFilterForm:
+    @pytest.fixture
+    def form(self, mocker):
+        return forms.NewsSearchForm(queryset=mocker.Mock())
+
+    def test_filter_by_search_term(self, mocker, form):
+        articles = form.queryset
+        articles.filter.return_value = "qs"
+        clean_data = {
+            "search": "great",
+        }
+        form.cleaned_data = clean_data
+        result = form.process()
+
+        assert result == "qs"
+        articles.filter.assert_called_once_with(
+            Q(title__icontains="great") | Q(intro__icontains="great")
+        )
+
+    def test_filter_by_search_no_data(self, mocker, form):
+        articles = form.queryset
+        form.cleaned_data = {}
+        result = form.process()
+
+        assert result == articles
+
+    def test_filter_by_tags(self, mocker, form):
+        articles = form.queryset
+        articles.filter.return_value = "qs"
+        clean_data = {
+            "tags": "running",
+        }
+        form.cleaned_data = clean_data
+        result = form.process()
+
+        assert result == "qs"
+        articles.filter.assert_called_once_with(tags__in="running")
+
+    def test_filter_by_tags_no_data(self, mocker, form):
+        articles = form.queryset
+        form.cleaned_data = {}
+        result = form.process()
+
```

Comment:
And similarly here. Means that if your process changed to remove the if statement the test would fail. Whereas currently it would still continue to pass I think